kaldb/src/main/java/com/slack/kaldb/logstore/search/aggregations/UniqueCountAggBuilder.java

```diff
@@ -0,0 +1,80 @@
+package com.slack.kaldb.logstore.search.aggregations;
+
+import java.util.Objects;
+
+public class UniqueCountAggBuilder extends ValueSourceAggBuilder {
+  public static final String TYPE = "cardinality";
+  private final Long precisionThreshold;
+
+  public UniqueCountAggBuilder(String name, String field, Object missing, Long precisionThreshold) {
+    super(name, field, missing);
+
+    this.precisionThreshold = precisionThreshold;
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  public Long getPrecisionThreshold() {
+    return precisionThreshold;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
I can start adding dedicated tests for the equals/hashcodes for these aggs. They do have coverage already in the serialize tests and proto tests, but those are only happy-path tests.

I'm not sure I'm going to keep this equals/hashcode approach of keeping them on the individual classes. While there are a few pros to this approach, I'm not sure the risk of missing a field makes it worth it.

Reply:
This will be addressed in the followup PR, #499. I already have started reworking the equals/hashfunctions there as I needed to add additional classes to support pipeline aggregations. I added a comment there referencing this discussion as well.